TESTS/netsocket/README.md

```diff
@@ -1654,6 +1669,311 @@ through a dedicated socket
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname_async()` in a row 6 times with a different host names. At least one host name shall be found from cache.
+2.  Verify that callback is called with correct parameters 6 times.
+
+**Expected result:**
+
+Callback is called with NSAPI_ERROR_OK and IP address 6 times.
+
+### ASYNCHRONOUS_DNS_CACHE
+
+**Description:**
+
+Verify that the caching of DNS results works correctly. Call `NetworkInterface::gethostbyname_async()` 5 times with the same host name and verify result after each request. For first request, cache shall not contain the host name. Verify that first request completes slower than the requests made after it (where the response is found from cache).
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname_async()` with a host name. For first request, host name shall not be found from cache.
+2.  Verify that callback is called with correct parameters.
+3.  Repeat the sequence 4 times using the same host name.
+4.  For each request, calculate how long time it takes for DNS query to complete.
+
+**Expected result:**
+
+Callback is called with NSAPI_ERROR_OK and IP address 5 times. First request shall complete slower than the requests made after it (where the response is found from cache).
+
+### ASYNCHRONOUS_DNS_NON_ASYNC_AND_ASYNC
+
+**Description:**
+
+Verify that non-asynchronous i.e. blocking DNS queries and asynchronous i.e. non-blocking queries work at the same time. Call `NetworkInterface::gethostbyname_async()`. Right after that make 6 non-asynchronous `NetworkInterface::gethostbyname()` calls with different host names.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname_async()` with a host name. Host name shall not be found from cache.
+2.  Call `gethostbyname()` 6 times with a different host names (none of the names shall be same as in step 1).
+3.  Verify that each `gethostbyname()` returns success.
+4.  Verify that the asynchronous callback is called with correct parameters.
+
+**Expected result:**
+
+All operations shall return NSAPI_ERROR_OK and IP address.
+
+### ASYNCHRONOUS_DNS_CANCEL
+
+**Description:**
+
+Verify that asynchronous DNS query cancel works correctly. Call `NetworkInterface::gethostbyname_async()` in a row 6 times with a different host names. Cache shall contain 3 host names used in requests. This can be achieved e.g. by running test "Asynchronous DNS non-asynchronous and asynchronous" before this test and using same host names in this run. For each request that was given an unique id, call cancel. Verify that callback is not called for cancelled requests.
```

Comment:
Please globally delete the hyphen in "nonasynchronous".
Please globally capitalize "ID".
Please globally delete the extra "l" in "canceled".